spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BaseRowReader.java

```diff
@@ -48,6 +50,17 @@ abstract class BaseRowReader<T extends ScanTask> extends BaseReader<InternalRow,
     super(table, taskGroup, tableSchema, expectedSchema, caseSensitive);
   }
 
+  BaseRowReader(
+      Table table,
+      ScanTaskGroup<T> taskGroup,
+      Schema tableSchema,
+      Schema expectedSchema,
+      boolean caseSensitive,
+      Integer pushedLimit) {
+    this(table, taskGroup, tableSchema, expectedSchema, caseSensitive);
+    this.pushedLimit = pushedLimit;
```

Comment:
Why do we push it all the way to the reader instead of simply reading just enough files to match the limit based on the number of records we have in the metadata for each data file?

Reply:
My initial thought is that we can use `ExpressionUtil$selectsPartitions` to test if the filter selects entire files. If so, we can simply find matching files so that the sum of `recordCount` exceeds the pushed limit. It may be more complicated for filters that only select some records within files. Let me explore this PR more for that.

Reply:
I think pushing it all the way to the reader could yield a bit more performance gain.

Actually if filter exists, Spark side doesn't push down limit.